topics/metabolomics/tutorials/msPurity_training/tutorial.md

```diff
@@ -0,0 +1,1006 @@
+
+So, the first one is a query SQLite database which should contain the *"unknown"* spectral database generated by the **msPurity.createDatabase {% icon tool %}** tool as part of the data processing workflow.  
+The second database is the spectral database of the library must contain the *“known”* spectra coming from public resources or generated by the users. The SQLite library database defaults to MoNA data, including Massbank, HMDB, LipidBlast and GNPS. A larger database can be downloaded from here??????????.
+  
+To create a user generated library SQLite database the following tool can be used to generate a SQLite database from a collection of MSP files: `msp2db` ([see on Pypi](https://pypi.org/project/msp2db/)). Unfortunatelly, this tool is not available on W4M Galaxy yet. It should be noted though, that as long as the schema of the spectral-database is as described here, then any database can be used for either the library or query - even allowing for the same database to be used. 
+  
+The spectral matching functionality includes four main components : 
+  - spectral filtering : spectral filtering simply filters both the library and the query spectra to search (for example, choosing the source of the library, the instrument, the retention time, the PPM tolerance of the precursor, etc.).
+  - spectral alignment : this step involves aligning the query peaks with the library peaks. The approach used is similar to the modified pMatch algorithm described in Zhou et al 2015.
+  - spectral matching : the spectral matching of the aligned spectra is performed against a combined intensity and a weighted vector m/z - created for the query and library spectra (wq and wl). See below: ```w = intensity^x ∗ mz^y```. Where x and y represent the weight factors, by default to `x = 0.5 and y = 2` according to MassBank. These can however be adjusted by the user. The aligned weighted vectors are then matched using the cosine of the dot product, the cosine of the inverse dot product, and the composite dot product. See below for the cosine equation of the dot product. ```dpc = wq ∗ wl / √(∑wq^2) ∗ √(∑wl^2)```.
+  - summary of results  
+
+> ### {% icon hands_on %} Hands-on : msPurity.spectralMatching {% icon tool %}
+>
+> Like the previous tool, we don't use it during our tutorial but we present it if you need to go further in your researches. So you need the following parameters to be able to run **msPurity.spectralMatching {% icon tool %} tool** : 
+> {: .text-justify}
+>   - **Query spectra input and filters** : it corresponds to the database which will compare to the library database.
+>     - **Input** : you have 3 options here, keep the first one selected :
+>       - `SQLite database of (LC)-MS/MS data` : it is the database you created during **msPurity.spectralMatching {% icon tool %} tool** processing.
+>       - *Locally configured SQLite, MySQL or PostgreSQL database* : ????????????????????????????????????????
```

Comment:
@Tomnl  gave me this link https://github.com/computational-metabolomics/mspurity-galaxy/blob/master/tools/msPurity/dbconfig.R where there is an example of database to be able to build our own. But on Galaxy there is not yet the parameter to enter our own database when we select this option... See this here https://github.com/computational-metabolomics/mspurity-galaxy/issues/35
Should I complete this option with something like 
```
you can add a local database when you select this option. Please build it conform to this https://github.com/computational-metabolomics/mspurity-galaxy/blob/master/tools/msPurity/dbconfig.R
```